Plans step redesign: Implement the header section of each plan

niranjan-uma-shankar opened the pull request:
#### Changes proposed in this Pull Request

This PR implements the experiment that redesigns the signup flow plans step. Users in the treatment group will see the new header section for each plan in the signup plans step.

NOTE: 
1. This PR uses the changes in https://github.com/Automattic/wp-calypso/pull/49877 as the base branch and is built on top of it.
2. This PR only implements the header section. The rest of the plan sections are implemented in subsequent PRs.

**Important links**
1. [Figma design](https://www.figma.com/file/gnYBUOBthTMckJRJr3pZmk/Control-Experiments?node-id=1362%3A813)
2. pau2Xa-2yE-p2 for the project thread.


**Annual plans header:**

<img width="1655" alt="Screenshot 2021-02-19 at 11 01 54 AM" src="https://user-images.githubusercontent.com/1269602/108462060-f293b680-72a1-11eb-91d4-95118acfe83d.png">


**Monthly plans header:**

<img width="1658" alt="Screenshot 2021-02-19 at 11 02 02 AM" src="https://user-images.githubusercontent.com/1269602/108462079-f9222e00-72a1-11eb-98ad-85c9bf5e780d.png">

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Open http://calypso.localhost:3000/ and assign yourself to treatment of the `signup_plans_step_redesign_v1` experiment in the Abacus tool.
* Begin signup at `/start`and verify that the plans step UI matches the [Figma design](https://www.figma.com/file/gnYBUOBthTMckJRJr3pZmk/Control-Experiments?node-id=1362%3A813). Verify for both free subdomain and custom domain selections.
* Verify that the UI matches the Figma design for both annual and monthly plan views.
* Confirm that in Calypso's /plans view, the UI is unchanged. 
* Also confirm that the launch flow UI is unchanged.
* Verify that `control` experience is unchanged. 

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/wp-calypso/pull/49877

niranjan-uma-shankar committed:
Query experiments to set the variation for this experiment

niranjan-uma-shankar committed:
Display header and sub-header based on the experiment variation

niranjan-uma-shankar committed:
CSS changes

niranjan-uma-shankar committed:
Render the features comparison component if the user is in the treatment group

niranjan-uma-shankar committed:
Create the plan feature comparison component

niranjan-uma-shankar committed:
UI changes to the header of the plan features component

niranjan-uma-shankar committed:
UI changes to match the figma

a8ci18n commented on client/my-sites/plan-features-comparison/header.jsx:
```diff
@@ -0,0 +1,430 @@
+			const annualPriceObj = getCurrencyObject( rawPriceAnnual, currencyCode );
+			const annualPriceText = `${ annualPriceObj.symbol }${ annualPriceObj.integer }`;
+
+			return translate( 'billed as %(price)s annually', {
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **3** times:
`translate( 'billed as %s annually' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/feature-comparison-signup-plans)

niranjan-uma-shankar committed:
Show the savings from paying annually

niranjan-uma-shankar committed:
Show the savings from paying annually

niranjan-uma-shankar committed:
Show popular badge as per design

niranjan-uma-shankar committed:
Move Experiment assignment to the plans step

niranjan-uma-shankar committed:
Add QueryExperiments in the main component

niranjan-uma-shankar committed:
Merge branch 'add/new-plans-layout-signup' into add/feature-comparison-signup-plans

niranjan-uma-shankar committed:
Merge branch 'trunk' into add/new-plans-layout-signup

niranjan-uma-shankar committed:
Merge branch 'add/new-plans-layout-signup' into add/feature-comparison-signup-plans

niranjan-uma-shankar committed:
Cleanup

dzver reviewed (commented):
I followed the test plan and I observed the following:

1. The launch-site flow in currencies where first-year promotional pricing is present look a bit unexpected:

<img width="1341" alt="Screenshot 2021-02-16 at 19 38 33" src="https://user-images.githubusercontent.com/82778/108102941-e9cc9600-7091-11eb-93fa-32c38492a1b7.png">

vs this in Trunk:

<img width="1328" alt="Screenshot 2021-02-16 at 20 01 08" src="https://user-images.githubusercontent.com/82778/108102964-ef29e080-7091-11eb-9485-4e48a38a0c00.png">


2. The mobile view is missing in treatment

<img width="353" alt="Screenshot 2021-02-16 at 19 32 49" src="https://user-images.githubusercontent.com/82778/108103005-fbae3900-7091-11eb-8e0d-0914ee8756b1.png">

There's some logic for discounts, is it exactly the same as the existing logic?

niranjan-uma-shankar committed:
Change subbheader to remove the risk free purchase copy per feedback received

niranjan-uma-shankar committed:
Hide blink of header text when the experiment is still loading

niranjan-uma-shankar committed:
Merge branch 'add/new-plans-layout-signup' into add/feature-comparison-signup-plans

niranjan-uma-shankar committed:
Enter into the experiment only for desktop resolutions

niranjan-uma-shankar committed:
Hide first year promotional pricing for treatment

niranjan-uma-shankar committed:
Marge from base branch

niranjan-uma-shankar committed:
Fix CSS issue seen on first year promotional pricing in control

niranjan-uma-shankar commented:
> The launch-site flow in currencies where first-year promotional pricing is present look a bit unexpected:

Nice catch, I've fixed this.

Also, I've removed the promotional pricing striked out UI for treatment in these currencies, because the "You're saving ...." copy shown for annual prices doesn't fit well when the promotional pricing is shown. Looks like we hide the striked out UI in new onboarding flow too, so keeping the same behaviour here.

<img width="1655" alt="Screenshot 2021-02-19 at 11 01 54 AM" src="https://user-images.githubusercontent.com/1269602/108462342-76e63980-72a2-11eb-88b7-52a1cccc1070.png">

dzver reviewed (approved):
OK, but in case we launch in INR/PHP/MXN this will have to be revisited.

The issue is that there's no mention that the renewal price will be different. The checkout also seems to suffer from this problem so the actual price will not be visible anywhere.

<img width="587" alt="Screenshot 2021-02-19 at 18 19 45" src="https://user-images.githubusercontent.com/82778/108531159-0e697d80-72df-11eb-8174-64827df9d915.png"> The actual cost is 4200. I don't see it anywhere.

So I think this design is only fine because we'll launch the experiment for en/us, which will see USD pricing.

niranjan-uma-shankar committed:
Plans step redesign: Implement the CTA button (#50005)

* Action button

* Plans redesign: Available and missing features (#50040)

* Show available and missing features in a side-by-side layout

* Show notice that certain monthly features are only available with annual plans

* Remove commented out lines, and remove an unnecessary if condition

* Force prettier

* Make the unavailable features in a light shade of gray; use a different version of unlimited premium themes feature

* Design feedback: Move the "included in annual plans" copy above the feature

* Remove missing features from the UI

* Fix bug that shows a JS error if a free subdomain is selected

* Fix included in annual plans notice

* Mention live chat support as 24x7 for business and ecommerce plans

* Add some padding for the last feature in the last column

* Plans redesign: Add new toggle design (#50227)

* Implement the new toggle UI

* Add a hover effect for the toggle buttons

* i18n bot recommendation

* Revert
